### PR TITLE
Add slugs to joblistings and articles and use them in common places

### PR DESCRIPTION
**Pull request by ollfkaih**
# Description

- Fix the infinite loop when redirecting from /{id}
- Add slugs to joblistings and articles
- Use slugs in common links and on frontpage

# Result

No visual changes except for url

# Testing

- [x] I have thoroughly tested my changes.
Seems to work fine now

---

Resolves ABA-212

### CODE REVIEW

File: app/routes/articles/ArticleDetailRoute.ts [hunk 1]
```diff
@@ -36,6 +44,11 @@ const mapStateToProps = (state, props) => {
     });
   });
   const emojis = selectEmojis(state);
+  setTimeout(() => {
+    if (article?.slug && articleIdOrSlug !== article.slug) {
+      props.history.replace(`/articles/${article.slug}`);
+    }
+  }, 0);
```
Comment (ivarnakken) [inline app/routes/articles/ArticleDetailRoute.ts:51]: Did you try out `useHistory`?

This is quite hacky, but still better than nothing. I'd like a comment here to explain why the timeout is needed though.

**Review by ivarnakken — APPROVED** [review]
Cool! 🏅

File: app/routes/articles/ArticleDetailRoute.ts (hunk 1 above)
Comment (ollfkaih) [reply app/routes/articles/ArticleDetailRoute.ts:51]: Read the rules of hooks (See https://reactjs.org/link/invalid-hook-call for tips about how to debug and fix this problem).
useHistory cannot be used here.

I can add an explainer.

Comment (ivarnakken) [reply app/routes/articles/ArticleDetailRoute.ts:51]: Yes it wouldn't work in the route of course, but why not move it to the page component?

Comment (ollfkaih) [reply app/routes/articles/ArticleDetailRoute.ts:51]: Well, that might work just as well, I can do some experimentation with it

Comment (ivarnakken) [reply app/routes/articles/ArticleDetailRoute.ts:51]: Since it's just a `replace` I don't really mind where it is

Comment (ollfkaih) [reply app/routes/articles/ArticleDetailRoute.ts]: I've moved it for Joblistings and Articles, Events will have to stay like this until EventDetail is rewritten to a function component

**Review by LudvigHz — APPROVED** [review]

**Pull request merged**